algosdk/future/transaction.py

```diff
@@ -2393,7 +2393,7 @@ def verify(self, message):
                 try:
                     verify_key.verify(message, subsig.signature)
                     verified_count += 1
-                except BadSignatureError:
+                except (BadSignatureError, ValueError, TypeError):
```

Comment:
I'm guessing that most users simply want to know that `verify()` has failed, deal with it, and won't really care why this has occurred (at least not in their runtime handling). For such users, swallowing these two exceptions makes life easier. 

But could there be something about these different errors that could allow users to have more nuanced recovery strategies? I kind of doubt it, but it's worth knowing before we just add the blanket conversion from error to False.

Reply:
Thinking more about it, `ValueError` and `TypeError` sounds like they could be generated by bad user input/usage. If that is the case, then we really wouldn't want to swallow these errors.

Reply:
Good point - I also thought about this and couldn't reach a conclusive decision. Our current unit tests do seem to test the `ValueError` here by testing the wrong number of bytes: https://github.com/algorand/py-algorand-sdk/blob/f0c9430ea6fda8e47ddcadd0579d36f5c66317e0/test_unit.py#L2694 

I'm not exactly sure how we want `verify` to behave in these cases so I wanted other people's inputs as well. If we choose to include this change, I can also add some test cases to check the two new error types.

Reply:
Summarizing a verbal with @algochoi, @tzaffi, and @jasonpaulos:
* We agreed the expedient path is to keep the PR's behavior and add unit tests validating behavior.
* We left with a loose agreement that revisiting the return type out-of-band is worthwhile.  For example, moving from boolean to a richer data type will simplify disambiguating failure cases.

As of https://github.com/algorand/py-algorand-sdk/pull/309/commits/c3b94a251444affa826a98a2a38eada8df86b1eb, I think the PR matches what we verbally agreed to.  Since I didn't open the thread, I'll leave as is without resolving.